src/librustc/traits/mod.rs

```diff
@@ -138,7 +138,7 @@ pub enum ObligationCauseCode<'tcx> {
 
     /// In an impl of trait `X` for type `Y`, type `Y` must
     /// also implement all supertraits of `X`.
-    ItemObligation(DefId),
+    ItemObligation(DefId, Option<Span>),
 
     /// Like `ItemObligation`, but with extra detail on the source of the obligation.
     BindingObligation(DefId, Span),
```

Comment:
Wait, looks like a version with a `Span` already exists, or is it too restricted to work for you?

Or I guess `BindingObligation` itself could be replaced with `ItemObligation` now.

Reply:
At the very least seeing this tells me that having a `Span` in `ObligationCauseCode` is probably fine.

Reply:
#69745 acknowledges this and uses `BindingObligation` instead. Some of the output is slightly worse, but can be iterated on.